src/main/updater.ts

```diff
@@ -22,12 +22,12 @@ export class Updater {
     });
   }
 
-  check() {
-    if (isDev) return;
-    this.autoUpdater.checkForUpdates();
+  check(): Promise<UpdateCheckResult | null> {
+    if (isDev) return Promise.resolve(null);
+    return this.autoUpdater.checkForUpdates();
```

Comment:
Why this change is needed?

Reply:
`AppUpdater #checkForUpdates` returns Promise, but old code omit the result.
https://github.com/electron-userland/electron-builder/blob/v29.30.0/packages/electron-updater/src/AppUpdater.ts#L215

Reply:
That looks correct, but you don't use a return value of this method, right?

https://github.com/bdash-app/bdash/blob/2285efaff60601511b2f5f8a83536980dff20c66/src/main/updater.ts#L31

So I think that there is no problem with the original code.

It look good in my local.

```diff
diff --git a/src/main/updater.ts b/src/main/updater.ts
index f444f5c..0998f1b 100644
--- a/src/main/updater.ts
+++ b/src/main/updater.ts
@@ -1,4 +1,4 @@
-import { autoUpdater, AppUpdater, UpdateCheckResult } from "electron-updater";
+import { autoUpdater, AppUpdater } from "electron-updater";
 import isDev from "electron-is-dev";
 import logger from "./logger";
 
@@ -22,12 +22,12 @@ export class Updater {
     });
   }
 
-  check(): Promise<UpdateCheckResult | null> {
-    if (isDev) return Promise.resolve(null);
-    return this.autoUpdater.checkForUpdates();
+  check() {
+    if (isDev) return;
+    this.autoUpdater.checkForUpdates();
   }
 
-  async watch() {
+  watch() {
     this.check();
     setInterval(() => this.check(), WATCH_INTERVAL);
   }
```

```
$ yarn run lint
yarn run v1.19.2
$ yarn run lint:eslint && yarn run lint:tsc && yarn run lint:prettier
$ eslint --ext '.js,.jsx,.ts,.tsx' .
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: ~3.2.1

YOUR TYPESCRIPT VERSION: 3.7.2

Please only submit bug reports when using the officially supported version.

=============
$ tsc --noEmit
$ prettier --list-different ./**/*.{js,jsx,ts,tsx,css,json}
   Done in 12.50s.
```

Reply:
Yep, TS 3.7 can compile original code.
But it is not good to ignore the result of method `check`.

For example, rewrite check:
```js
check() {
  //if (isDev) return;
  this.autoUpdater.checkForUpdates();
}
```

```console
$ yarn start
yarn run v1.19.2
$ electron app/development
21:42:29.200 › Checking for update
21:42:29.235 › TypeError: Cannot read property 'catch' of undefined
    at App.<anonymous> (/Users/user/work/js/bdash/app/development/out/main.js:136:30)
    at App.emit (events.js:208:15)
GVA info: Successfully connected to the Intel plugin, offline Gen95
21:42:29.606 › Error: Error: ENOENT: no such file or directory, open '/Users/user/work/js/bdash/app/development/dev-app-update.yml'
(node:43858) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/Users/user/work/js/bdash/app/development/dev-app-update.yml'
(node:43858) UnhandledPromiseRejectionWarning: Error: ENOENT: no such file or directory, open '/Users/user/work/js/bdash/app/development/dev-app-update.yml'
(node:43858) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:43858) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:43858) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:43858) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Reply:
In fact, my change is not enough yet.
I rollback to original code, and write it later.

Reply:
rollback https://github.com/bdash-app/bdash/pull/62/commits/89381594a267e6409b0ba739d3bb0c3134713197